code/workspaces/auth-service/src/permissions/permissions.js

```diff
@@ -31,36 +31,48 @@ const projectifyPermissions = ({ projectKey, permissions }) => {
   return permissions.map(permission => prefix.concat(permission));
 };
 
-//    [ { role: 'instanceAdmin', permissions: [/see permissions.yml/] } ]
-// -> [ ['system:instance:admin', ...] ]
-const systemifyPermissions = ({ permissions }) => {
-  const prefix = `${SYSTEM}${roleDelim}`;
-  return permissions.map(permission => prefix.concat(permission));
-};
-
 const flattenArray = (previous, current) => {
   if (Array.isArray(current)) {
     return [...previous, ...current];
   }
   return previous;
 };
 
+function getInstanceAdminPermissions(instanceRole) {
+  if (!instanceRole) {
+    return [];
+  }
+  const { permissions } = permissionAttributes[INSTANCE_ADMIN_ROLE_KEY];
+  return permissions.map(permission => `${SYSTEM}${roleDelim}${INSTANCE}${roleDelim}${permission}`);
+}
+
+function getCataloguePermissions(catalogueRole) {
+  if (!catalogueRole) {
+    return [];
+  }
+  const permissions = permissionAttributes[CATALOGUE_ROLE_KEY]
+    .filter(role => role.role === catalogueRole)
+    .map(role => role.permissions)
+    .reduce((allPermissions, permissionsArray) => [...allPermissions, ...permissionsArray], [])
```

Comment:
Could this be done in one using `flatMap` ([reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap))?

Reply:
Yes - updated my node version to v14.15.1, to match https://github.com/NERC-CEH/datalab/blob/master/code/api.Dockerfile#L18.